smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/impl/ReactiveKafkaConsumer.java

```diff
@@ -124,7 +127,16 @@ public void setRebalanceListener() {
     Uni<ConsumerRecords<K, V>> poll() {
         if (polling.compareAndSet(false, true)) {
             return runOnPollingThread(c -> {
-                return c.poll(pollTimeout);
+                if (System.getSecurityManager() == null) {
```

Comment:
The security manager can't change after startup right? So maybe we can capture that in a constant?

Reply:
Apparently, it can change if the previous one allows it :) BTW I just saw that from Java 17 Security Manager is deprecated for removal: https://openjdk.java.net/jeps/411

Reply:
@darranl Do you know anything about this removal of the SM?

Reply:
The use of the security manager remains fully supported in the next LTS release of Java, at this stage it is deprecated for later removal but there are ongoing discussions regarding how it is being handled and API compatibility issues.